### PR TITLE
Implement API endpoints and tests for workflow sharing options.

### PR DESCRIPTION
**Pull request by jmchilton**
First step toward replacing the sharing.mako template and web controller and for being able to write API tests around users shared with functionality for workflows.

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/managers/workflows.py [hunk 1]
```diff
@@ -349,6 +355,27 @@ def serialize_workflow_invocations(self, invocations, **kwd):
 CreatedWorkflow = namedtuple("CreatedWorkflow", ["stored_workflow", "workflow", "missing_tools"])
 
 
+class WorkflowSerializer(sharable.SharableModelSerializer):
+    """
+    Interface/service object for serializing pages into dictionaries.
```
Comment (davelopez) [inline lib/galaxy/managers/workflows.py]:
```suggestion
    Interface/service object for serializing workflows into dictionaries.
```

File: lib/galaxy/webapps/galaxy/services/workflows.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+from galaxy.managers.workflows import (
+    WorkflowSerializer,
+    WorkflowsManager,
+)
+from galaxy.webapps.galaxy.services.base import ServiceBase
+from galaxy.webapps.galaxy.services.sharable import ShareableService
+from lib.galaxy.managers.workflows import WorkflowSerializer
```
Comment (davelopez) [inline lib/galaxy/webapps/galaxy/services/workflows.py]:
```suggestion

```

File: lib/galaxy/managers/workflows.py [hunk 3]
```diff
@@ -349,6 +355,27 @@ def serialize_workflow_invocations(self, invocations, **kwd):
...
+    model_manager_class = WorkflowsManager
+    SINGLE_CHAR_ABBR = "w"
+
+    def __init__(self, app: MinimalManagerApp):
+        super().__init__(app)
+        self.page_manager = WorkflowsManager(app)
+
+        self.default_view = "summary"
+        self.add_view("summary", [])
+        self.add_view("detailed", [])
```
Comment (davelopez) [inline lib/galaxy/managers/workflows.py]: Does this probably need adjustments?

**Review by davelopez — COMMENTED** [review]
Great! I thought it would require more changes to get workflows working with the common sharing system :)

**Review by bgruening — APPROVED** [review]
LGTM. But has WIP in the title. Maybe @jmchilton has more plans with this one?

**Review by davelopez — APPROVED** [review]

**Comment by jmchilton** [comment]
Thanks for the quick turn around y'all!

**Pull request merged**